### PR TITLE
Move Wallet service to separate local crate

### PR DESCRIPTION
**Pull request by jovfer**

### CODE REVIEW

File: libindy/indy-api-types/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+    #[macro_export]
+    macro_rules! secret {
+        ($val:expr) => {{ $val }};
+    }
+
+    #[cfg(not(debug_assertions))]
+    #[macro_export]
+    macro_rules! secret {
+        ($val:expr) => {{ "_" }};
+    }
```
Comment (Artemkaaas) [inline libindy/indy-api-types/src/lib.rs]: I would rather move it to indy-utils

File: libindy/indy-utils/src/crypto/base64/rust_base64.rs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-use errors::prelude::*;
+use indy_api_types::errors::prelude::*;
```
Comment (Artemkaaas) [inline libindy/indy-utils/src/crypto/base64/rust_base64.rs]: I like moving errors to separate crat. We will avoid cross dependencies and duplication in `indy-api-types` and `indy-utils`

File: libindy/indy-utils/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+[package]
+name = "indy-utils"
+version = "0.1.0"
+authors = ["Sergey Minaev <sergey.minaev@dsr-corporation.com>"]
```
Comment (Artemkaaas) [inline libindy/indy-utils/Cargo.toml]:
```suggestion
authors = ["Hyperledger Indy Contributors <hyperledger-indy@lists.hyperledger.org>"]
```

File: libindy/src/services/indy-wallet/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+[package]
+name = "indy-wallet"
+version = "0.1.0"
+authors = ["Sergey Minaev <sergey.minaev@dsr-corporation.com>"]
```
Comment (Artemkaaas) [inline libindy/src/services/indy-wallet/Cargo.toml]:
```suggestion
authors = ["Hyperledger Indy Contributors <hyperledger-indy@lists.hyperledger.org>"]
```

File: libindy/indy-utils/src/crypto/base64/rust_base64.rs (hunk 2 above)
Comment (jovfer) [reply libindy/indy-utils/src/crypto/base64/rust_base64.rs]: duplication/cross dependencies fixed by moving next_handle_* helpers to utils

**Pull request merged**